docs/developers_notes/basis_module.md

```diff
@@ -0,0 +1,130 @@
+2. `_get_samples`: Returns a tuple of equidistant samples (as flat numpy arrays) for each dimension of the basis function. These samples are used to form a grid over which the basis is evaluated.
+3. `_check_n_basis_min`: Checks the minimum number of basis functions required. This requirement can be specific to the type of basis.
+
+
+## The `AdditiveBasis` Class
+
+This class extends the abstract `Basis` class and represents a set of basis functions that are combined additively. It overrides the `_evaluate`, `_get_samples`, and `_check_n_basis_min` methods with specific implementations for additive basis functions.
+
+### Overridden Methods
+
+1. `_evaluate`: For an `AdditiveBasis` object, this method should evaluate each basis function at the specified samples and then add the results together.
+2. `_get_samples`: This method should return equidistant samples over the domain of each basis function. The samples should then be summed together.
+3. `_check_n_basis_min`: This method checks whether the number of basis functions meets the minimum requirement for an additive basis set.
+
+## The `MultiplicativeBasis` Class
+
+This class extends the `Basis` abstract class and represents a set of basis functions that are combined multiplicatively. It overrides the `_evaluate`, `_get_samples`, and `_check_n_basis_min` methods with specific implementations for multiplicative basis functions.
+
+### Overridden Methods
+
+1. `_evaluate`: For a `MultiplicativeBasis` object, this method should evaluate each basis function at the specified samples and then multiply the results together.
+2. `_get_samples`: This method should return equidistant samples over the domain of each basis function. The samples should then be multiplied together.
+3. `_check_n_basis_min`: This method checks whether the number of basis functions meets the minimum requirement for a multiplicative basis set.
+
+## The `SplineBasis` and `MSplineBasis` Classes
+
+The `SplineBasis` class is an abstract subclass of `Basis` that represents a spline basis function. The `MSplineBasis` class is a concrete subclass of `SplineBasis` that implements a specific type of spline basis, the M-spline.
+
+### Overridden Methods for `MSplineBasis`
+
+1. `_evaluate`: For an `MSplineBasis` object, this method should evaluate the M-spline at the specified samples.
+2. `_get_samples`: This method should return equidistant samples over the domain of the M-spline.
+3. `_check_n_basis_min`: This method checks whether the number of M-spline basis functions meets the minimum requirement.
+
+## The `RaisedCosineBasis`, `RaisedCosineBasisLinear`, and `RaisedCosineBasisLog` Classes
+
+The `RaisedCosineBasis` class is an abstract subclass of `Basis` that represents a raised cosine basis function. The `RaisedCosineBasisLinear` and `RaisedCosineBasisLog` classes are concrete subclasses of `RaisedCosineBasis` that implement specific types of raised cosine basis functions.
```

Comment:
needs a longer description or a link to what a "raised cosine basis function" means (with reference, I think). Also, what's shared among them that makes it worth having their own subclass, and how they differ.

E.g., "The two concrete subclasses have the same `_evaluate` function but differ in their `_get_samples` method"